p4-16/psa/examples/psa-example-parser-error-handling.p4

```diff
@@ -0,0 +1,601 @@
+        }
+    }
+    state copy_clone_e2e_meta_parser_error {
+        meta.clone_reason = clone_e2e_meta.clone_reason;
+        meta.to_cpu_error_hdr = clone_e2e_meta.to_cpu_error_hdr;
+        // See comment in copy_clone_i2e_meta_parser_error.
+        transition accept;
+    }
+    state clone_e2e_unknown_format_id {
+        verify(false, error.UnknownCloneE2EFormatId);
+    }
+
+    state packet_in_parsing {
+        p.apply(buffer, hdr, meta);
+        transition accept;
+    }
+}
+
+control egress(inout headers hdr,
+               inout metadata meta,
+               in    psa_egress_input_metadata_t  istd,
+               inout psa_egress_output_metadata_t ostd)
+{
+    apply {
+        if (meta.clone_reason == CloneReason_t.PARSER_ERROR) {
+            // No headers have been parsed in the egress parser, so
+            // the entire packet is payload.  Create a header to
+            // prepend in front of the packet and send it to the CPU.
+            hdr.to_cpu_error_hdr = meta.to_cpu_error_hdr;
+
+            // Note that while this packet was received by the egress
+            // pipeline as a result of a clone operation, it is being
+            // sent via the PSA "normal" packet path to the CPU port.
+            // It will thus go through the normal egress deparser
+            // processing, where the hdr.to_cpu_error_hdr will be
+            // emitted first, followed by no other valid headers
+            // (because we did not parse any headers for packets
+            // passing through the 'true' branch of this 'if'
+            // statement), followed by the payload, which should be
+            // the entire packet as it most recently arrived at the
+            // egress parser.
+            exit;
+        }
+        if (istd.parser_error != error.NoError) {
+            // Try handling egress parser errors similarly to ingress
+            // parser errors.
+            //
+            // Note: Unlike the ingress case, here the contents of the
+            // cloned packet will be that output by the egress
+            // deparser when this egress control block is finished, so
+            // if this program is running on a P4 implementation where
+            // emitting the valid headers that exist after the egress
+            // parser encounters an error is _not_ the same as the
+            // packet sent into the egress parser, then the control
+            // plane will not receive exactly the same packet that
+            // caused the parse error.
+            handle_parser_errors.apply(istd.parser_error,
+                istd.packet_path, istd.egress_port,
+                meta.clone_reason, meta.to_cpu_error_hdr);
+            meta.to_cpu_error_hdr.ingress = 0;
+            egress_drop(ostd);
+            ostd.clone = true;
+            ostd.clone_session_id = PSA_CLONE_SESSION_TO_CPU;
+            exit;
+        }
+    }
+}
+
+// BEGIN:Compute_New_IPv4_Checksum_Example
```

Comment:
copy & paste?

Reply:
Removed the 2 unused obsolete comments you pointed out.  This new example program is currently not used anywhere in the PSA.mdk file.  Not sure whether we want to do so for this program, or perhaps merely mention its name and a brief description in the document?